rust/src/storage/azure/mod.rs

```diff
@@ -126,25 +182,75 @@ fn to_storage_err(err: Box<dyn Error + Sync + std::marker::Send>) -> StorageErro
 }
 
 fn to_storage_err2(err: azure_storage::core::Error) -> StorageError {
+    if let azure_storage::core::Error::CoreError(azure_core::Error::Other(ref other_err)) = err {
+        match other_err.downcast_ref::<azure_core::error::Error>() {
+            Some(other_err) => match other_err.downcast_ref::<azure_core::error::HttpError>() {
+                Some(e) => {
+                    if e.status() == 404 {
+                        return StorageError::NotFound;
+                    }
+                }
+                None => {}
+            },
+            None => {}
+        }
+    }
     StorageError::AzureStorage { source: err }
 }
 
+async fn list_obj_future(
```

Comment:
nit: maybe add a quick todo comment here, that we want to move away from this once, Pageable is sent in the azure sdk.

Reply:
done!